src/Control/Carrier/Error/Either.hs

```diff
@@ -11,62 +11,37 @@ module Control.Carrier.Error.Either
 , run
 ) where
 
-import Control.Applicative (Alternative(..), liftA2)
+import Control.Applicative (Alternative(..))
 import Control.Carrier
 import Control.Effect.Error
 import Control.Monad (MonadPlus(..), (<=<))
 import qualified Control.Monad.Fail as Fail
 import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
+import Control.Monad.Trans.Except
 
 -- | Run an 'Error' effect, returning uncaught errors in 'Left' and successful computations’ values in 'Right'.
 --
 --   prop> run (runError (pure a)) === Right @Int @Int a
 runError :: ErrorC exc m a -> m (Either exc a)
-runError = runErrorC
+runError = runExceptT . runErrorC
 
-newtype ErrorC e m a = ErrorC { runErrorC :: m (Either e a) }
-  deriving (Functor)
+newtype ErrorC e m a = ErrorC { runErrorC :: ExceptT e m a }
+  deriving (Applicative, Functor, Monad, Fail.MonadFail, MonadFix, MonadIO, MonadTrans)
 
-instance Applicative m => Applicative (ErrorC e m) where
-  pure a = ErrorC (pure (Right a))
-  {-# INLINE pure #-}
-  ErrorC f <*> ErrorC a = ErrorC (liftA2 (<*>) f a)
-  {-# INLINE (<*>) #-}
-
-instance Alternative m => Alternative (ErrorC e m) where
-  empty = ErrorC empty
+instance (Alternative m, Monad m) => Alternative (ErrorC e m) where
```

Comment:
We still define our own `Alternative` and `MonadPlus` instances in order to retain our existing semantics instead of adopting `ExceptT`’s (which handles them itself using a `Monoid` constraint on `e` instead of passing them along to `m`).

Reply:
Can we leave this as documentation in the file? Haddocks on those instances, maybe?

Reply:
Good call 👍